cmds/cpio/pkg/newc/newc.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// newc implements the interface for new type cpio files.
+package newc
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/u-root/u-root/cmds/cpio/pkg"
+)
+
+const (
+	newcMagic = "070701"
+	magicLen  = len(newcMagic)
+	headerLen = 13*8 + magicLen
+)
+
+type Header struct {
+	Ino      uint64
+	Mode     uint64
+	UID      uint64
+	GID      uint64
+	Nlink    uint64
+	Mtime    uint64
+	FileSize uint64
+	Major    uint64
+	Minor    uint64
+	Rmajor   uint64
+	Rminor   uint64
+	NameSize uint64
+	_        uint64
+}
+
+func (f *Header) String() string {
+
+	return fmt.Sprintf("Ino %d Mode %#o UID %d GID %d Nlink %d Mtime %#x FileSize %d Major %d Minor %d RMajor %d Rminor %d NameSize %d",
+		f.Ino,
+		f.Mode,
+		f.UID,
+		f.GID,
+		f.Nlink,
+		// what a mess. This fails on travis.
+		//time.Unix(int64(f.Mtime), 0).String(),
+		f.Mtime,
+		f.FileSize,
+		f.Major,
+		f.Minor,
+		f.Rmajor,
+		f.Rminor,
+		f.NameSize)
+}
+
+type newcReader cpio.CpioReader
+type newcWriter cpio.CpioWriter
+
+func Header2Info(h *Header, n string, i *cpio.Info) {
```

Comment:
I would leave that for the optimizer -- that is hopefully basic enough for Go to realize that it shouldn't be passed on the stack.